src/Templates/MethodTemplate.cshtml

```diff
@@ -29,8 +29,29 @@ func (client @(Model.Owner)) @(Model.MethodSignature) (@Model.MethodReturnSignat
 @EmptyLine
 </text>
 }
-
-    req, err := client.@(Model.PreparerMethodName)(@(Model.HelperInvocationParameters(complete: false)))
+@if (Model.IsPageable && !Model.IsLongRunningOperation() && !Model.IsNextMethod)
+{
+    var fnField = Model.ReturnType.Body.Cast<CompositeTypeGo>().UnwrapPageType().FnFieldName;
+    var nextMethod = Model.NextMethod;
+    if (nextMethod != null)
+    {
+        var returnType = nextMethod.MethodReturnType;
+        var nextLinkField = Model.ReturnType.Body.Cast<CompositeTypeGo>().UnwrapPageType().NextLink;
+    <text>
+    result.@fnField = func(lastResult @returnType) (@returnType, error) {
+        if lastResult.@nextLinkField == nil || len(to.String(lastResult.@nextLinkField)) < 1 {
+            return @returnType{}, nil
+        }
+        return client.@(nextMethod.Name)( @Model.HelperInvocationParameters(false), *lastResult.@nextLinkField )
```

Comment:
>HelperInvocationParameters(false) [](start = 49, length = 33)

What is the purpose of the `Complete` parameter ? I saw that is always `false`. When would it be `true` ? . Also nit: would leave the named parameter `complete: value` and put a comment on the method to make it more clear what it does.

Reply:
Hmm it's possible that in some older code that has since been removed that somebody needed this to be set to true.  I can remove the parameter as it appears to be useless.